src/node.cc

```diff
@@ -4265,35 +4265,17 @@ void Init(int* argc,
 }
 
 
-struct AtExitCallback {
-  AtExitCallback* next_;
-  void (*cb_)(void* arg);
-  void* arg_;
-};
-
-static AtExitCallback* at_exit_functions_;
-
-
-// TODO(bnoordhuis) Turn into per-context event.
 void RunAtExit(Environment* env) {
-  AtExitCallback* p = at_exit_functions_;
-  at_exit_functions_ = nullptr;
-
-  while (p) {
-    AtExitCallback* q = p->next_;
-    p->cb_(p->arg_);
-    delete p;
-    p = q;
-  }
+  env->RunAtExit();
 }
 
 
+thread_local Environment* thread_local_env;
```

Comment:
* Should this be `static`?
* I think this doesn’t need an explicit initializer, but would you mind adding `= nullptr` just for clarity?